consensus/core/src/block.rs

```diff
@@ -167,13 +216,78 @@ impl SignedBlock {
 
 /// Verifiied block allows access to its content.
 #[allow(unused)]
-#[derive(Deserialize, Serialize)]
+#[derive(Clone, Deserialize, Serialize)]
 pub(crate) struct VerifiedBlock {
-    pub block: Block,
+    pub data: Block,
     pub signature: bytes::Bytes,
 
     #[serde(skip)]
     serialized: bytes::Bytes,
 }
 
+impl BlockAPI for VerifiedBlock {
```

Comment:
ah I've used another way to get access to the underlying block struct on my PR https://github.com/MystenLabs/sui/pull/15684/files#diff-e00cbe61a5ac738cad4a7d2732a9a90f6dd6666e8e1eecd7d3ab77797375ca61R228 when dereferencing so we don't need to implement the `BlockAPI` methods for the `VerifiedBlock` and `SignedBlock`, but we essentially get a transparent access to `data`. Let's decide which one we want to go with @arun-koshy @mwtian .

Reply:
`Deref` seems pretty convenient, but a bit too powerful. Let's not implement `Deref` for `SignedBlock`. The point is to limit access before verifications. Otherwise we can use that for `VerifiedBlock` for now.